src/plugins/pcr/src/InSilicoPcrTask.h

```diff
@@ -22,31 +22,33 @@
 #ifndef _U2_IN_SILICO_PCR_TASK_H_
 #define _U2_IN_SILICO_PCR_TASK_H_
 
+#include <U2Algorithm/BaseTempCalc.h>
 #include <U2Algorithm/FindAlgorithmTask.h>
 
 #include <U2Core/GObjectReference.h>
 #include <U2Core/Task.h>
 
 namespace U2 {
 
-class InSilicoPcrTaskSettings {
-public:
-    InSilicoPcrTaskSettings();
+struct InSilicoPcrTaskSettings {
+    InSilicoPcrTaskSettings() = default;
+    ~InSilicoPcrTaskSettings() = default;
+    Q_DISABLE_COPY_MOVE(InSilicoPcrTaskSettings)
```

Comment:
Will making 'temperatureCalculator' a shared pointer help to keep the settings a plain data structure with no copy-magic?

Reply:
Yes, it will, I added this restriction because I added a usual pointer as a member, and if you have a pointer member, you should overall copy-move operators-constructors and a destructor - otherwise it could cause memory-pointing errors (deletion is a sort of it).